### DISCUSSION TITLE
How to Keep Messages Always Visible Above Modals?

### DISCUSSION THREAD

**Discussion by Cinnabar-HgS** [NG-ZORRO/ng-zorro-antd#8724] [Q&A]
I am using both modal and message components simultaneously. Initially, when I click to show the modal, the message appears above the modal. However, after closing the modal and clicking to show it again, the message ends up displaying beneath the modal, meaning the message gets obscured by the modal. How can I make sure the message always displays above the modal, regardless? I have tried modifying the z-index of both components, but it doesn't seem to work.

**Comment by Cinnabar-HgS** [comment]
and the version of ng-zorro-antd is 15.0.1.

**Comment by ParsaArvanehPA** [comment]
@Cinnabar-HgS Hi,
could you please provide a reproducable sample? 

The z-index solution that you mentioned should defenitly work, I can help you figure out a solution better if you give me a producable sample.

**Reply by Cinnabar-HgS** [reply]
thank you very much for your reply. I've discovered a new issue.

Since the entire project cannot be uploaded to GitHub, I created a small demo, but I was unable to reproduce the issue where my message were being obscured by the modal. At first, I suspected it was a conflict between components, but now it seems like there are some other issues within my project that are causing this bug, and even using z-index cannot resolve it.

For some reasons, the original project cannot be uploaded online, so I'll first try to see if I can solve this problem myself.

thank you again for your reply.

**Reply by ParsaArvanehPA** [reply]
> thank you very much for your reply. I've discovered a new issue.
> 
> Since the entire project cannot be uploaded to GitHub, I created a small demo, but I was unable to reproduce the issue where my message were being obscured by the modal. At first, I suspected it was a conflict between components, but now it seems like there are some other issues within my project that are causing this bug, and even using z-index cannot resolve it.
> 
> For some reasons, the original project cannot be uploaded online, so I'll first try to see if I can solve this problem myself.
> 
> thank you again for your reply.

Your welcome,
btw i meant more like a stackblitz link, I created one for you, but was not able to reproduce the issue you mentioned([link](https://stackblitz.com/edit/angular-hmyyzv?file=src%2Fapp%2Fapp.component.ts)).
let me know if I could be of more help 👍
